File: CSharp/DcfSegment.cs

namespace CSharp {
    internal class DcfSegment {
        public Position Start { get; }
        public Position End { get; }
        public string ClustId { get; }
        public double Length { get; }

        public DcfSegment(Position start, Position end, string clustId) {
            Start = start;
            End = end;
            ClustId = clustId;
            Length = start.GetDistance(end);
        }
    }
}

File: Java/src/main/java/com/loggersoft/performance/DcfSegment.java
package com.loggersoft.performance;

public class DcfSegment {
    private final Position start;
    private final Position end;
    private final String clustId;
    private final double length;

    public DcfSegment(Position start, Position end, String clustId) {
        this.start = start;
        this.end = end;
        this.clustId = clustId;
        length = start.getDistance(end);
    }

    public Position getStart() {
        return start;
    }

    public Position getEnd() {
        return end;
    }

    public String getClustId() {
        return clustId;
    }

    public double getLength() {
        return length;
    }
}
